packages/sdk-communication-layer/src/services/RemoteCommunication/ConnectionManager/initCommunicationLayer.ts

```diff
@@ -99,10 +99,12 @@ export function initCommunicationLayer({
     title = state.dappMetadata.name;
   }
 
+  const defaultDappId =
+    state.dappMetadata?.name ?? state.dappMetadata?.url ?? 'n/a';
   let dappId =
```

Comment:
this should be enough to cover all possibilities:
```const dappId = (typeof window !== ‘undefined’ && typeof window.location !== ‘undefined’ && window.location.hostname) || defaultDappId;```

also you don't need the `if(!dappId)` statement